pingcap/tiflow
# validator(dm): support start from time & fix & integration test

### What problem does this PR solve?


Issue Number: close #5449

### What is changed and how it works?
**this pr is large most due to change to `.proto` and test**

- support `validation start --start-time`
- refactor and add more check to `validation start`, `validation start` can be used in 2 ways:
  - enable validator on the fly(like create+start in task operation), in this case all validator of target subtask should not have been enabled before;
  - start stopped validator, all validator of target subtask should have been enabled, and cannot add explicit mode and start-time flag
- add `mode` to `validation status`
- fix bug that we set an not-completely-initialized subtaskConfig into subtask on validator start. now we only update needed config.
- fix bug that we initialized validator when expect stage is stopped(happens on failover), which may cause resource leak.
- extract real binlog from relay log `binlog pos`
- reset FirstValidateTS when loading pending row changes to avoid failed row changes be marked as error row immediately if validator stopped for a long time.
- add more integration test

### Check List 

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

#### Questions 

##### Will it cause performance regression or break compatibility?

##### Do you need to update user documentation, design documentation or monitoring documentation?

### Release note 

```release-note
`None`.
```
